extensions/oidc-db-token-state-manager/runtime/src/main/java/io/quarkus/oidc/db/token/state/manager/runtime/OidcDbTokenStateManagerRunTimeConfig.java

```diff
@@ -0,0 +1,33 @@
+package io.quarkus.oidc.db.token.state.manager.runtime;
+
+import static io.quarkus.runtime.annotations.ConfigPhase.RUN_TIME;
+
+import java.time.Duration;
+
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.smallrye.config.ConfigMapping;
+import io.smallrye.config.WithDefault;
+
+@ConfigMapping(prefix = "quarkus.oidc-db-token-state-manager")
+@ConfigRoot(phase = RUN_TIME)
+public interface OidcDbTokenStateManagerRunTimeConfig {
+
+    /**
+     * Whether Quarkus should delete expired token states.
+     * You can disable this feature when you want only one of your service instances to delete expired tokens.
+     */
+    @WithDefault("true")
+    boolean deleteExpired();
+
+    /**
+     * How often should Quarkus check for expired tokens. By default, the Quarkus will check every 20 seconds.
+     */
+    @WithDefault("20")
+    Duration deleteExpiredDelay();
+
+    /**
+     * Whether Quarkus should attempt to create database table where the token state is going to be stored.
+     */
+    @WithDefault("true")
+    boolean createDatabaseTableIfNotExists();
```

Comment:
Is it something that is already possible to configure at the Hibernate level ?

Reply:
Certainly, but this extension intentionally doesn't rely on Hibernate until Reactive and ORM work together. IMO we can revise it in the future.

Reply:
OK, let me ask another question, when would users choose not to create the table if it already does not exist ?

Reply:
My thoughts for showing how would Hibernate entity look for DB token state table was that it establishes contract - shows exactly what columns are needed.

> OK, let me ask another question, when would users choose not to create the table if it already does not exist ?

It's similar to strategy Hibernate schema strategy drop-and-create, you wouldn't use it in a production. Here, nothing bad will happen when we attempt to create table, so no harm. <======== to this point I could agree with you but:

FULL DISCLOSURE: this is my opinion from practice, I have zero statistics to support my belief

Usually you don't want to configure datasource with user that has rights for DDL operations like creating database tables. Such operations are done by DBA or someone with different credentials than are credentials provided to application. Application will usually only need DML (select ,...). So what I saw was that you have scripts to create database etc. that are executed prior application starts. And in such cases, you definitely want to create table yourself in a production.

Reply:
@michalvavrik Makes sense